microprocessor_simulator.py

```diff
@@ -118,30 +124,31 @@ def execute_instruction(self, instruction):
                 elif opcode == 'grt':
                     self.cond = REGISTER[ra.lower()] > REGISTER[rb.lower()]
                 elif opcode == 'add':
-                    REGISTER[ra.lower()] = f'{int(REGISTER[rb.lower()], 16) + int(REGISTER[rc.lower()], 16):02x}'
+                    REGISTER[ra.lower()] = convert_to_hex(int(REGISTER[rb.lower()], 16) + int(REGISTER[rc.lower()], 16),
+                                                          2)
                 elif opcode == 'sub':
-                    REGISTER[ra.lower()] = f'{REGISTER[rb.lower()] - REGISTER[rc.lower()]:02x}'
+                    REGISTER[ra.lower()] = convert_to_hex(REGISTER[rb.lower()] - REGISTER[rc.lower()], 2)
```

Comment:
REGISTER[ra.lower()] = convert_to_hex(REGISTER[rb.lower()] - REGISTER[rc.lower()], 8)

Do the same for the subsequent method calls